Promise/Promise+Extras.swift

```diff
@@ -73,19 +99,30 @@ public enum Promises {
             }).then(fulfill).catch(reject)
         })
     }
-
+    
+    /// Runs a block of code that creates a `Promise` but which can also `throw`.
+    /// - Parameter block: Code to generate a `Promise`; `throw`s are rejected as expcted.
+    /// - Returns: `Promise<T>` that is created from the given block
```

Comment:
```suggestion
    /// Runs a block of code that creates a `Promise` but which can also `throw`.
    /// Useful when setting up a promise with code that needs to throw.
    /// - Parameter block: Code to generate a `Promise`; `throw`ing creates a rejected promise with the thrown error.
    /// - Returns: `Promise<T>` that is created from the given block
```
(Is the `block` label too long now? I wasn't sure how to wrap it)